test/helpers/kubectl.go

```diff
@@ -172,10 +172,21 @@ func (kub *Kubectl) ExecPodCmd(namespace string, pod string, cmd string, options
 	return kub.Exec(command, options...)
 }
 
-// ExecPodCmdContext executes command cmd in background in the specified pod residing
+// ExecPodCmdContext synchronously executes command cmd in the specified pod residing in the
+// specified namespace. It returns a pointer to CmdRes with all the output.
+func (kub *Kubectl) ExecPodCmdContext(ctx context.Context, namespace string, pod string, cmd string, options ...ExecOptions) *CmdRes {
```

Comment:
Maybe there's a simpler approach. Looking over this, I actually see that this is likely to break nightly tests too so I better iterate on this a little more.

Reply:
I reworked it a little to launch all of the commands in the background then wait for the commands to complete and dump the output afterwards.